register.go

```diff
@@ -373,3 +373,48 @@ func configureRHSM(serverURL string) error {
 
 	return nil
 }
+
+// Call D-bus to get the server hostname from rhsm
+func getServerHostname() (string, error) {
+
+	var sname string
+
+	conn, err := dbus.SystemBus()
+	if err != nil {
+		return sname, fmt.Errorf("cannot connect to system D-Bus: %w", err)
+	}
+
+	if err := conn.Object(
+		"com.redhat.RHSM1",
+		"/com/redhat/RHSM1/Config").Call(
+		"com.redhat.RHSM1.Config.Get",
+		dbus.Flags(0),
+		"server.hostname",
+		"").Store(&sname); err != nil {
+		return "", unpackRHSMError(err)
+	}
+
+	return sname, nil
+}
+
+// Call D-bus to get the CA directory from rhsm
+func getCADir() (string, error) {
```

Comment:
This function and `getServerHostname` feel like they could be combined into a generic function that takes a config name and return argument:

```go
func getRHSMConfigOption(name string, val interface{}) error {
    conn, err := dbus.SystemBus()
    obj := conn.Object("com.redhat.RHSM1", "/com/redhat/RHSM1/Config")
    obj.Call("com.redhat.RHSM1.Config.Get", dbus.Flags(0), name, "").Store(val)
    return nil
}
```

Reply:
Done, that makes more sense.